### PR TITLE
SDL behaviour if VR not available

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
In case if VR interface respond with available = false. SDL should process VR related requests according to rules described in CRQ [APPLINK-20918 VR interface: SDL behavior in case HMI does not respond to IsReady_request or respond with "available" = false](https://adc.luxoft.com/jira/browse/APPLINK-20918)

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
Review proceed in #745

**Comment by LuxoftAKutsan** [comment]
@AGaliuzov please merge

File: src/components/application_manager/include/application_manager/hmi_interfaces_impl.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_IMPL_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_IMPL_H_
+#include <map>
+#include "application_manager/hmi_interfaces.h"
+#include "utils/macro.h"
+
+namespace application_manager {
+class HmiInterfacesImpl : public HmiInterfaces {
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/hmi_interfaces_impl.h]: @LuxoftAKutsan How about doxygen comments for the class itself and for all its methods

Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/hmi_interfaces_impl.h]: @AGaliuzov  f17cb6631086e6a139bcbff5adfe99ab843c782b

File: src/components/application_manager/include/application_manager/commands/mobile/register_app_interface_request.h [hunk 2]
```diff
@@ -87,6 +87,23 @@ class RegisterAppInterfaceRequest : public CommandRequestImpl {
...
+   * @param function_id interface specific ChangeRegistration
+   * @param language language of registration
+   * @param app_id application to change registration
+   */
+  void SendChangeRegistration(const hmi_apis::FunctionID::eType function_id,
+                              const int32_t language,
+                              const uint32_t app_id);
+
+  /**
+   * @brief SendChangeRagistrationOnHMI send required SendChangeRagistration n
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/commands/mobile/register_app_interface_request.h]: @LuxoftAKutsan looks like redundant symbol

File: src/components/application_manager/include/application_manager/hmi_interfaces.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_H_
+
+#include "interfaces/HMI_API.h"
+
+namespace application_manager {
+
+class HmiInterfaces {
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/hmi_interfaces.h]: @LuxoftAKutsan how about doxygen description for the class and its methods

File: src/components/application_manager/include/application_manager/hmi_interfaces_impl.h [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_IMPL_H_
+#include <map>
+#include "application_manager/hmi_interfaces.h"
+#include "utils/macro.h"
+#include "utils/lock.h"
+
+namespace application_manager {
+
+/**
+ * @brief The HmiInterfacesImpl class handles
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/hmi_interfaces_impl.h]: @LuxoftAKutsan looks too brief to me

File: src/components/application_manager/src/commands/hmi/ui_is_ready_response.cc [hunk 5]
```diff
@@ -53,6 +53,12 @@ void UIIsReadyResponse::Run() {
   HMICapabilities& hmi_capabilities = application_manager_.hmi_capabilities();
 
   hmi_capabilities.set_is_ui_cooperating(is_available);
+  const HmiInterfaces::InterfaceState inteface_state =
```
Comment (AGaliuzov) [inline src/components/application_manager/src/commands/hmi/ui_is_ready_response.cc]: @LuxoftAKutsan typo - have you meant `interface_state`?

File: src/components/application_manager/include/application_manager/hmi_interfaces_impl.h (hunk 4 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/hmi_interfaces_impl.h]: @AGaliuzov but it rely does not do anything else

File: src/components/application_manager/src/commands/hmi/ui_is_ready_response.cc (hunk 5 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/hmi/ui_is_ready_response.cc]: @AGaliuzov 9df2535

File: src/components/application_manager/include/application_manager/hmi_interfaces_impl.h (hunk 4 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/hmi_interfaces_impl.h]: @AGaliuzov but it does not do anything else

File: src/components/application_manager/include/application_manager/hmi_interfaces.h (hunk 3 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/hmi_interfaces.h]: @AGaliuzov  29e68de

File: src/components/application_manager/include/application_manager/commands/mobile/register_app_interface_request.h (hunk 2 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/mobile/register_app_interface_request.h]: @AGaliuzov 9df2535

**Pull request merged**